File: android/app/src/main/java/com/connect/club/react/bridge/iosLike/ui/UserSpeakerMicrophoneIconsView.kt
package com.connect.club.react.bridge.iosLike.ui

import android.content.Context
import android.view.View
import androidx.core.view.isVisible
import com.facebook.react.views.view.ReactViewGroup
import org.json.JSONObject

class UserSpeakerMicrophoneIconsView(context: Context?) : ReactViewGroup(context) {
    private var icSpeakerOn: View? = null

    fun onState(state: JSONObject) {
        val isExpired = state.getBoolean("isExpired")
        val isReaction = state.getString("reaction") != "none"
        val isSpeaker = state.getBoolean("isSpeaker")
        val isAbsoluteSpeaker = state.getBoolean("isAbsoluteSpeaker")
        val isLocal = state.getBoolean("isLocal")
        val audio = state.getBoolean("audio")

        isVisible = (isAbsoluteSpeaker || isSpeaker) && !isReaction && !isExpired

        if (isLocal) {
            if (isSpeaker && !isAbsoluteSpeaker) {
                AppManager.radarView?.hide()
            } else {
                AppManager.radarView?.show()
            }
        }
        if (!isVisible) return

        icSpeakerOn?.isVisible = audio
    }

    override fun addView(child: View?, index: Int) {
        super.addView(child, index)

        if (child?.contentDescription == "icSpeakerOn") {
            icSpeakerOn = child
            child.isVisible = false
        }
    }
}

File: ios/ReactNativeModules/ui/UserSpeakerMicrophoneIconsView.swift
//
//  UserSpeakerMicrophoneIconsView.swift
//  connectreactive
//
//  Created by Sergei Golishnikov on 13/05/2021.
//

import Foundation
import UIKit

@objc(UserSpeakerMicrophoneIconsView)
final class UserSpeakerMicrophoneIconsView: UIView {

    override init(frame: CGRect) {
        super.init(frame: frame)
        accessibilityIdentifier = "speakerMicrophoneIcon"
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func onState(_ state: JSON) {
        let isExpired = state.isExpired.bool == true
        let isSpeaker = state.isSpeaker.bool == true
        let isAbsoluteSpeaker = state.isAbsoluteSpeaker.bool == true

        isVisible = isSpeaker && !isExpired

        let isLocal = state.isLocal.bool == true
        if isLocal {
            if isSpeaker && !isAbsoluteSpeaker {
                AppManager.instance.radarView?.hide()
            } else {
                AppManager.instance.radarView?.show()
            }
        }
    }

    override func addSubview(_ view: UIView) {
        super.addSubview(view)
        if view.accessibilityLabel == "icSpeakerOn" {
            view.isHidden = true
        }
    }
}
